axlearn/common/attention.py

```diff
@@ -1808,11 +1828,18 @@ def _compute_attention(
             k_proj: [batch_size, source_length, num_heads, per_head_dim].
             v_proj: [batch_size, source_length, num_heads, per_head_dim].
             attention_logit_biases: See ``On attention logit biases`` in the file comments.
+            segment_ids: See ``segment_ids`` in the file comments.
 
         Returns:
             The context of shape [batch_size, target_length, num_heads, per_head_dim],
             and probs of shape [batch, num_heads, target_length, source_length].
         """
+        if segment_ids is not None:
+            raise ValueError(
+                "segment_ids is not supported. To use segment_ids, construct "
+                "attention_logit_biases using an AttentionLogitBiasLayer."
+            )
```

Comment:
Here we can merge or convert `segment_ids` into `attention_logit_biases`. This allows the caller to pass only `segment_ids=..., attention_logit_biases=None` into MultiheadAttention.